api/acl.go

```diff
@@ -634,6 +634,32 @@ func (a *ACL) PolicyRead(policyID string, q *QueryOptions) (*ACLPolicy, *QueryMe
 	return &out, qm, nil
 }
 
+// PolicyReadByName retrieves the policy details including the rule set with name.
+func (a *ACL) PolicyReadByName(policyName string, q *QueryOptions) (*ACLPolicy, *QueryMeta, error) {
+	r := a.c.newRequest("GET", "/v1/acl/policy/name/"+url.QueryEscape(policyName))
```

Comment:
Unfortunately this endpoint does not yet exist. I think adding it would be good but it will require additions in numerous other locations.

Off the top of my head all of the following would be needed:

agent/http_oss.go (register the new endpoint similar to what we do for roles)
agent/acl_endpoint.go (add the new endpoint func to read the role by id)
agent/structs/acl.go - add a `PolicyIDIsName bool` field to indicate in the RPC requests that the policy should be retrieve by name instead of by ID.
agent/consul/acl_endpoint.go - Update the `PolicyRead` function to take into account the new field indicating to read by name and when set to `true` use `state.ACLPolicyGetByName` instead of `state.ACLPolicyGetByID` for retrieving the policy.

Reply:
Excellent. Thanks @mkeeler for the quick review. I'll go ahead and the missing segments to make sure it actually functions.

Reply:
I've gone ahead and added everything you requested except the `PolicyIDIsName` boolean you stated. I didn't see the reason why it would be required here as the `ACLPolicyReadByName` would only be retrieved when doing a direct GET request to the `/acl/policy/name`. I looked over the role and didn't see any boolean logic for the definition there either. 

Most likely I am missing something here. Can you kindly explain briefly over the reasoning to adding a boolean to the `ACLPolicyGetRequest` struct?

Reply:
@abaez I dont think its necessary with having a second string to store the name in the request struct. In other places we use a single string + boolean to indicate the mode of query but we can just as easily check if one is the empty string or not.

Reply:
@mkeeler so the main reason why I went with the second string was due to trying to imitate the same work done on role struct when identifying the name or id. I wanted to keep it as consistent as possible, instead of fully implementing this in a different approach.

Reply:
That makes total sense and is good with me. 

Unfortunately the API isn't terribly consistent throughout but it was a good decision to keep it consistent here.